GD32F4XX/standard_peripheral/Include/gd32f4xx_timer.h

```diff
@@ -624,7 +624,7 @@ void timer_deinit(uint32_t timer_periph);
 /* initialize TIMER init parameter struct */
 void timer_struct_para_init(timer_parameter_struct* initpara);
 /* initialize TIMER counter */
-void timer_init(uint32_t timer_periph, timer_parameter_struct* initpara);
+void __timer_init(uint32_t timer_periph, timer_parameter_struct* initpara);
```

Comment:
I don't agreed with this change at this moment. I think we should continue to follow initial proposal **timer_initialize**, which is already in tree. After discussion at PR #8 be finished, someone can send a PR and refactor.

The main purpose of #8 is create a document about how things should be addressed.

Reply:
I think that there should be a consistent way to handle name collisions (`__` was an idea, but can be something else). Creating a script would also help in adding/updating HALs. Related to this problem, I think GD32 HAL headers should not be part of `soc.h` since, unfortunately, it has a poor naming policy. A solution similar to STM32 LL HAL can be easily implemented, that is, generic HAL headers that are included as needed by drivers or applications. In LL case, the reason was different, though: to avoid thousands of static inlines being _pasted_ in almost every compilation unit.

Reply:
Hi @gmarull ,

I understand your point and I already anticipated this kind of problem, that is why I opened #8. If we merge this as is right now we got an inconsistent API because we will add two different name conventions. I prefer keep consistency between API and discuss how to evolve at #8. After we decide the right path, we apply at once the whole solution.

Reply:
I think the prefix '__' is a more simple and explicit way to fix the name conflict.

Reply:
The old conversion will be fix at this PR.

Reply:
I have update other libraries conflict name.

Reply:
Yes, do not use double underscore. This is a problem for C++ too.

https://stackoverflow.com/questions/3325815/what-does-a-double-underscore-mean-in-a-variable-name-in-the-c-language#3325829

Keep current code convention and then let's define at #8 best way to move forward.

Reply:
maybe `gd32_` and `GD32_` prefixes then? CAN definitions need to be renamed and there's no established convention, so I think we should just solve it here and move on.

Reply:
I'm OK to have GD32_ prefix in all defines/enums.

For function names, my preference is add gd32_ for all functions, not strong opinion, just a preference. I have been thinking this could be the solution in the long term. In this case, I suggest below two options:

1) Change to long name timer_initialize (current at hal_gigadevice)
2) Add gd32_timer_init prefix (need normalize [1])

I understand if we use [1] we have space to continue discussion and plan how move forward. If decision is add gd32_ prefix my understanding is that decision was taken.